Client/Frontend/Browser/Authenticator.swift

```diff
@@ -48,6 +73,25 @@ class Authenticator {
         return self.promptForUsernamePassword(viewController, credentials: nil, protectionSpace: challenge.protectionSpace, loginsHelper: nil)
     }
 
+    private static func findValidLoginAmongstLogins(logins: Cursor<LoginData>, forProtectionSpace protectionSpace: NSURLProtectionSpace) -> LoginData? {
```

Comment:
Note that this is assuming that cursors can be iterated multiple times. I'd be more comfortable turning `logins` into a `[LoginData]` up on line 45 of `Authenticator.swift`. (Methods written against `SequenceType` won't care.)